src/data/matrix/basic.lean

```diff
@@ -518,8 +517,7 @@ instance semiring.is_scalar_tower [decidable_eq n] [monoid R] [distrib_mul_actio
 by { ext, apply dot_product_smul }
 
 /-- This instance enables use with `mul_smul_comm`. -/
-instance semiring.smul_comm_class [decidable_eq n] [monoid R] [distrib_mul_action R α]
-  [smul_comm_class R α α] :
+instance semiring.smul_comm_class [monoid R] [distrib_mul_action R α] [smul_comm_class R α α] :
```

Comment:
This was previously requires because `has_one` was needed by monoid.to_mul_action

Reply:
Thanks, I was almost at this conclusion :-)